### PR TITLE
feat(moe): add unified block-scale FP8 support

### PR DESCRIPTION
**Pull request by feih-nv**
## 📌 Description

Adds first-class DeepSeek FP8 and MXFP8 block-scale execution through the unified `MoELayer` API. These formats previously existed in the configuration schema and legacy flat APIs, but were not executable through `MoELayer`.

Fixed one issue in `tests/moe_ep/test_split_fused_moe_kernel_vs_reference.py`.

### What changed

- Added `TrtllmFp8BlockRunner` to the unified `MoELayer` API with:
  - DeepSeek FP8 and MXFP8 block-scale execution
  - Precomputed and in-kernel routing
  - Local expert offsets, autotuning, and CUDA graphs
- Added format-aware weight and activation preparation:
  - DeepSeek FP8: FP32 128-element/128x128 block scales
  - MXFP8: UE8M0 32-element scales with shuffled MajorK weights
- Fixed DeepSeek FP8 to preserve caller-provided activation scales.
- Restricted unified block-FP8 support to the validated SM100 family (SM100 and SM103).
- Added independent-reference, routing-replay, CUDA graph, and unified-fuzzer coverage.

## 🔍 Scope and follow-up

Unified FP8 support is split into two PRs because block-scale and per-tensor FP8 have distinct scaling and execution contracts. This PR covers block-scale FP8.

### Follow-up work for per-tensor FP8

- Add `TrtllmFp8PerTensorRunner`
- Add per-tensor weight and scalar-scale preparation
- Register `QuantVariant.FP8PerTensor` with `MoELayer`
- Initially support `FromLogits`, matching the legacy API
- Add independent conformance, CUDA-graph, tactic, fuzzer, and model-profile coverage

### Additional future work

- FP8 expert-parallel integration
- `do_finalize=False`
- Routing replay
- LoRA and fused shared experts
- OA parameters
- Non-SwiGLU activations

Existing legacy tests remain under the Mirror → Bridge → Prune policy. The legacy flat APIs are unchanged. The unified runner delegates to the existing TRTLLM `MoERunner` and kernel entry points.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

## 🧪 Tests

Validated on SM100:
- `11 passed` — unified block-FP8 conformance suite
- `19 passed` — unified API validation subset
- `2 passed` — curated DeepSeek FP8 and MXFP8 fuzzer cases with production autotuning
- `1 passed` — existing FP4 `FromLogits` fuzzer confirmation on SM100

## Summary by CodeRabbit

* **New Features**
  * Added unified TRT-LLM block-FP8 MoE support for **DeepSeek FP8** and **MXFP8**.
  * Added variant-aware block-FP8 weight/activation preparation and a new public **block-FP8 runner**.
  * Enabled both pre-routed and **logits-based** routing (including expert offsets), with CUDA graph capture/replay support.
* **Bug Fixes**
  * Tightened block-FP8 backend SM support to explicitly allowed targets.
  * Corrected FP8 block activation/scale handling for accurate kernel dispatch.
* **Tests**
  * Expanded block-FP8 conformance and fuzz coverage; updated backend support/validation expectations.

### CODE REVIEW

File: flashinfer/fused_moe/runners.py [hunk 1]
```diff
@@ -503,6 +503,293 @@ def __hash__(self):
...
+            )
+        elif routing_input_mode == RoutingInputMode.PackedPrecomputed:
+            _validate_prerouted_inputs(
+                act, num_tokens, routing.top_k, "TrtllmFp8BlockRunner"
+            )
+            routing_logits = None
+            routing_bias = None
+            weight_bits = (
+                act.topk_weights.to(torch.bfloat16).view(torch.int16).to(torch.int32)
+            )
```
[the remark this answers is not in the document]
Comment (feih-nv) [reply flashinfer/fused_moe/runners.py:784]: `view(dtype)` only requires contiguity when the element sizes differ. bf16→int16 is a same-width (2-byte) reinterpret, which PyTorch allows regardless of contiguity — verified on a non-contiguous bf16 tensor (both `.t()` and strided slices view fine). And for any non-bf16 input, the preceding `.to(torch.bfloat16)` already returns a contiguous copy. This also mirrors the existing FP4/BF16 routed runners, which use the identical expression, so I'll keep it as-is for consistency.

**Review by Anerudhan — APPROVED** [review]

**Review by aleozlx — APPROVED** [review]
